File: src/Unidesk/Unidesk/Dtos/EnumsDto.cs
using System.ComponentModel.DataAnnotations;

namespace Unidesk.Dtos;

public class EnumsDto : DtoBase
{
    [Required]
    public List<DepartmentDto> Departments { get; set; } = new();

    [Required]
    public List<FacultyDto> Faculties { get; set; } = new();

    [Required]
    public List<SchoolYearDto> SchoolYears { get; set; } = new();

    [Required]
    public List<ThesisOutcomeDto> ThesisOutcomes { get; set; } = new();

    [Required]
    public List<ThesisTypeDto> ThesisTypes { get; set; } = new();

    [Required]
    public List<StudyProgrammeDto> StudyProgrammes { get; set; } = new();
}

File: src/Unidesk/Unidesk.Client/src/api-client/models/EnumsDto.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { DepartmentDto } from './DepartmentDto';
import type { FacultyDto } from './FacultyDto';
import type { SchoolYearDto } from './SchoolYearDto';
import type { StudyProgrammeDto } from './StudyProgrammeDto';
import type { ThesisOutcomeDto } from './ThesisOutcomeDto';
import type { ThesisTypeDto } from './ThesisTypeDto';

export type EnumsDto = {
    readonly _DtoType?: string | null;
    departments?: Array<DepartmentDto> | null;
    faculties?: Array<FacultyDto> | null;
    schoolYears?: Array<SchoolYearDto> | null;
    thesisOutcomes?: Array<ThesisOutcomeDto> | null;
    thesisTypes?: Array<ThesisTypeDto> | null;
    studyProgrammes?: Array<StudyProgrammeDto> | null;
};
